src/main/scala/uk/gov/nationalarchives/tdr/api/metadatainputvalidation/MetadataInputTags.scala

```diff
@@ -0,0 +1,55 @@
+package uk.gov.nationalarchives.tdr.api.metadatainputvalidation
+
+import sangria.execution.BeforeFieldResult
+import sangria.schema.{Argument, Context}
+import uk.gov.nationalarchives.tdr.api.graphql.fields.FileMetadataFields.UpdateBulkFileMetadataInput
+import uk.gov.nationalarchives.tdr.api.graphql.{ConsignmentApiContext, ValidationTag}
+import uk.gov.nationalarchives.tdr.api.model.file.NodeType
+import uk.gov.nationalarchives.tdr.api.auth.AuthorisationException
+import uk.gov.nationalarchives.tdr.api.graphql.DataExceptions.InputDataException
+import uk.gov.nationalarchives.tdr.api.service.FileService.FileDetails
+
+import java.util.UUID
+import scala.concurrent._
+import scala.language.postfixOps
+
+trait MetadataInputTag extends ValidationTag
+
+case class ValidateMetadataInput[T](argument: Argument[T]) extends MetadataInputTag {
+
+  override def validateAsync(ctx: Context[ConsignmentApiContext, _])(implicit executionContext: ExecutionContext): Future[BeforeFieldResult[ConsignmentApiContext, Unit]] = {
+    val arg: T = ctx.arg[T](argument.name)
+
+    val inputFileIds: Seq[UUID] = arg match {
+      case input: UpdateBulkFileMetadataInput => input.fileIds
+    }
+    val inputConsignmentId: UUID = arg match {
+      case input: UpdateBulkFileMetadataInput => input.consignmentId
```

Comment:
We can get the from the `inputFileIds` and `inputConsignmentId` from a single matcher 

```
    val (inputFileIds, inputConsignmentId) = arg match {
      case input: UpdateBulkFileMetadataInput => (input.fileIds, input.consignmentId)
    }
```